patches/minecraft/net/minecraft/entity/monster/EntityZombie.java.patch

```diff
@@ -56,6 +56,15 @@
                              entityzombie.func_180482_a(this.field_70170_p.func_175649_E(new BlockPos(entityzombie)), (IEntityLivingData)null);
                              this.func_110148_a(field_110186_bp).func_111121_a(new AttributeModifier("Zombie reinforcement caller charge", -0.05000000074505806D, 0));
                              entityzombie.func_110148_a(field_110186_bp).func_111121_a(new AttributeModifier("Zombie reinforcement callee charge", -0.05000000074505806D, 0));
+@@ -394,7 +406,7 @@
+             entityzombievillager.func_82149_j(entityvillager);
+             this.field_70170_p.func_72900_e(entityvillager);
+             entityzombievillager.func_180482_a(this.field_70170_p.func_175649_E(new BlockPos(entityzombievillager)), new EntityZombie.GroupData(false));
+-            entityzombievillager.func_190733_a(entityvillager.func_70946_n());
++            entityzombievillager.setForgeProfession(entityvillager.getProfessionForge());
```

Comment:
Actually `getProfession` in these two classes are the only place where ids > 6 aren't supported. And since `getProfession` should still return ids <= 6 then the change above is still needed.